specs/enhance_gluster_cluster_sync_logic.adoc

```diff
@@ -0,0 +1,307 @@
+* Volume geo-rep details
+* Client connections per volume
+* Identification of arbiter volumes in cluster (if any)
+* Instead of `gstatus` use aggregations across bricks utilization to calculate
+volume utilization
+* Instead of `gstatus` use peer status and volume status to aggregate and
+calculate the cluster status
+
+
+== Proposed change
+
+* Use `get-state detail` and `get-state volumeoptions` commands instead of plain
+`get-state` as first provides the detailed (e.g. client connections etc.)
+information for the volume and second is option to list all the volume options
+with default values
+
+* Use `get-state` output to figure out the volume snapshot details and update in
+central store
+
+* Calculate the estimation time for re-balance job. This should be largest
+value among the `time_left` values reported by get-state from different nodes of
+the cluster
+
+* Use `get-state` output for geo-rep session details like no. of geo-rep
+sessions and active geo-rep sessions and update per volume in central store
+
+* Figure out the client connections per brick from `get-state` output and
+aggregate for volume and update in central store
+
+* Identify a arbiter volume and mark accordingly in central store, based on
+field `volume{n}.arbiter_count` in `get-state` output
+
+* Remove the uses of `gstatus` and use aggregations across bricks of the volume
+to decide the volume utilization data
+
+* Remove the uses of `gstatus` and use aggregations of `gluster peer status` and
+`gluster volume status` to decide the cluster health status
+
+=== Alternatives
+
+None
+
+=== Data model impact
+
+* Add an additional object `Snapshot` to hold volume snapshots details
+
+```
+Snapshot:
+  attrs:
+    vol_id:
+      help: Volume id
+      type: String
+    name:
+      help: name of the snapshot
+      type: String
+    id:
+      help: uuid of the snapshot
+      type: String
+    time:
+      help: Creation time of the snapshot
+      type: String
+    description:
+      help: description of snapshot
+      type: String
+    status:
+      help: status of snapshot
+      type: String
+  enabled: true
+  value: clusters/$TendrlContext.integration_id/Volumes/$Volume.vol_id/Snapshots/$Snapshot.id
+  list: clusters/$TendrlContext.integration_id/Volumes/$Volume.vol_id/Snapshots
+help: gluster volume snapshot details
+```
+
+* Add an additional field `rebal_estimation_time` under `RebalanceDetails` to
+hold the re-balance estimation time value
+
+* Add an additional object `GeoRepSessions` to hold the geo-rep session details
+for gluster volumes
+
+```
+TBD
+```
+
+* Add an additional field `client_connections` under `Volume` object to hold
+the client connections details
+
+* Add ad additional fields `client_connections` and `arbiter` under `Brick`
+object to hold the client connections and to mark if the brick is arbiter brick
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+* Provide an API to list snapshot details for volumes
```

Comment:
The APIs will soon be required, most likely in the [Gluster Monitoring milestone](https://github.com/Tendrl/specifications/milestone/1). Though, I suggest that for now, we exclude the API details from this spec and file a separate spec or an issue for API endpoints for all the additional gluster state information we now include as part of this spec.